src/helpers.php

```diff
@@ -27,13 +55,19 @@ function backpack_url($path = null)
      */
     function backpack_avatar_url($user)
     {
+        $placehold = 'https://placehold.it/160x160/00a65a/ffffff/&text='.$user->name[0];
+
         switch (config('backpack.base.avatar_type')) {
             case 'gravatar':
-                return Gravatar::fallback('https://placehold.it/160x160/00a65a/ffffff/&text='.$user->name[0])->get($user->email);
+                if (backpack_users_have_email()) {
+                    return Gravatar::fallback('https://placehold.it/160x160/00a65a/ffffff/&text='.$user->name[0])->get($user->email);
```

Comment:
You need to escape the `text=` or eventually somehow something will go wrong.

https://groups.google.com/forum/#!topic/comp.mail.headers/MDZPvKXTb3w

Reply:
I wondered the same thing one day, so tried a bunch of email addresses to break it, but wasn't able to. Apparently Gravatar pipes this through a pretty solid method ```buildUrl()``` that takes care of everything :-)

Reply:
Relying on outside services for security is likely to get us in trouble.

EquiFax could purchase the Gravatar guys, for example.

Reply:
That would be a treat :-) Gravatar _itself_ doesn't do the escaping. The Laravel package that makes the call to them does the escaping. If we escape it before that package escapes it, we'd just be escaping it twice.